hazelcast/src/test/java/com/hazelcast/internal/util/ThreadAffinityTest.java

```diff
@@ -0,0 +1,161 @@
+/*
+ * Copyright (c) 2008-2020, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.internal.util;
+
+import com.hazelcast.test.HazelcastParallelClassRunner;
+import com.hazelcast.test.annotation.ParallelJVMTest;
+import com.hazelcast.test.annotation.QuickTest;
+import org.jetbrains.annotations.NotNull;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+
+import java.util.BitSet;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertTrue;
+
+@RunWith(HazelcastParallelClassRunner.class)
+@Category({QuickTest.class, ParallelJVMTest.class})
+public class ThreadAffinityTest {
+
+    @Test
+    public void whenNull() {
+        ThreadAffinity threadAffinity = new ThreadAffinity(null);
+
+        assertFalse(threadAffinity.isEnabled());
+        assertNull(threadAffinity.nextAllowedCpus());
+    }
+
+    @Test
+    public void whenEmptyString() {
+        ThreadAffinity threadAffinity = new ThreadAffinity("");
+
+        assertFalse(threadAffinity.isEnabled());
+        assertNull(threadAffinity.nextAllowedCpus());
+    }
+
+    @Test(expected = ThreadAffinity.InvalidAffinitySyntaxException.class)
+    public void whenSyntaxError() {
+        new ThreadAffinity("abc");
+    }
+
+    @Test(expected = ThreadAffinity.InvalidAffinitySyntaxException.class)
+    public void whenTrailingComma() {
+        new ThreadAffinity("10,");
+    }
+
+    @Test(expected = ThreadAffinity.InvalidAffinitySyntaxException.class)
```

Comment:
Would make sense to add a test for negative numbers too. Both for groups and thread count.

Reply:
Negative integers are not parsed by the parser.

Reply:
Shouldn't we get an `InvalidAffinitySyntaxException` for such a setting?

Reply:
You will. Check the syntax for an integer. it will expect 1 digit followed by zero or more digits. It doesn't expect an `-`.